cicecore/cicedynB/dynamics/ice_dyn_evp.F90

```diff
@@ -954,37 +961,39 @@ subroutine evp (dt)
                   strairxE(:,:,iblk), strairyE(:,:,iblk), &
                   strocnxE(:,:,iblk), strocnyE(:,:,iblk))
 
-            ! If we are coupling to a C grid ocean model
-!           do ij =1, icelle
-!              i = indxei(ij,iblk)
-!              j = indxej(ij,iblk)
-
-!              strocnxT(i,j,iblk) = strocnxE(i,j,iblk) / aiE(i,j,iblk)
-!           enddo
-
-!           do ij =1, icelln
-!              i = indxni(ij,iblk)
-!              j = indxnj(ij,iblk)
-
-!              strocnyT(i,j,iblk) = strocnyN(i,j,iblk) / aiN(i,j,iblk)
-!           enddo
-
          enddo
          !$OMP END PARALLEL DO
 
       endif
 
-      call ice_HaloUpdate (strocnxT,           halo_info, &
+      ! strocn computed on U, N, E as needed. Map strocn U divided by aiu to T
+      ! TODO: This should be done elsewhere as part of generalization?
+      ! conservation requires aiu be divided before averaging
+      work1 = c0
+      work2 = c0
+      !$OMP PARALLEL DO PRIVATE(iblk,ij,i,j)
+      do iblk = 1, nblocks
+      do ij = 1, icellu(iblk)
+         i = indxui(ij,iblk)
+         j = indxuj(ij,iblk)
+         work1(i,j,iblk) = strocnx(i,j,iblk)/aiu(i,j,iblk)
+         work2(i,j,iblk) = strocny(i,j,iblk)/aiu(i,j,iblk)
+      enddo
+      enddo
+      !$OMP END PARALLEL DO
+      call ice_HaloUpdate (work1,              halo_info, &
```

Comment:
Why not send strocnx,y directly into ice_HaloUpdate?  Does that change the answers?

Reply:
The order of the divide and halo shouldn't matter.  We still need a temporary unless we are going to use strairxT as a temporary.  So the options are

work1=strocnx/aiu
call halo_update(work1)
call grid_average_X2Y(work1, strocnxT)

call halo_update(strocnx)
work1 = strocnx/aiu
call grid_average_X2Y(work1, strocnxT)

call halo_update(strocnx)
strocnxT = strocnx/aiu
call grid_average_X2Y(strocnxT, strocnxT)

We are doing the first.  The second is largely identical.  The third is something I would prefer not to do as it uses strocnxT as a temporary place holder for the "U" value and then averages in place, which is a feature I have deprecated because it's not needed and (in my opinion), can be confusing.